### PR TITLE
fix: correct Thin Plate Spline warp normalization for grid_sample

### LINKED ISSUE #3459: [Bug]: warp_image_tps returns all-zero output instead of warped image
### 🐛 Describe the bug

`kornia.geometry.transform.warp_image_tps` appears to return an image filled with zeros (black) instead of the expected warped image, even when using the example from the documentation.

### 🔄 Steps to Reproduce

```bash
1. Run the following code, it applies a Thin Plate Spline warp to a random image, but instead of producing a warped result, the function returns an output image where every pixel is zero.
```

### 💻 Minimal Code Example

```python
import kornia
import torch
from kornia.geometry.transform import get_tps_transform, warp_image_tps

points_src = torch.rand(1, 25, 2)
points_dst = torch.rand(1, 25, 2)
image = torch.rand(1, 3, 64, 64)

kernel_weights, affine_weights = get_tps_transform(points_dst, points_src)
warped = warp_image_tps(image, points_src, kernel_weights, affine_weights)
print(warped.unique())
```

### ✅ Expected behavior

The function should return a valid warped image tensor, i.e, non-zero pixel values, where the input image is transformed according to the Thin Plate Spline mapping.

### ❌ Actual behavior

The function returns an output image where all pixel values are zero, instead of the expected warped image.

### 🔧 Environment

```shell
- Kornia version:0.8.2
- PyTorch version:2.8.0
- Python version:3.9.6
- OS (e.g., Linux, macOS, Windows):macOS Sequoia 15.5(Apple Silicon M4)
- Installation method (pip, conda, from source):pip
- CUDA/cuDNN version (if applicable):NA
- GPU model (if applicable):Apple Silicon M-series
```

### 📝 Additional context

The above code block will fix this issue and produce expected output.

### 🤝 Contribution Intent

- [x] I plan to submit a PR to fix this bug
- [ ] I'm reporting this bug but not planning to fix it

### PR DESCRIPTION
**Pull request by dheeraj25406**
## 📝 Description

https://github.com/kornia/kornia/issues/3459

Fixes #3459 

Thin Plate Spline was generating warp grids in pixel coordinates, but `torch.nn.functional.grid_sample` expects coordinates normalized to the `[-1, 1]` range. Because of this mismatch, the sampling grid was incorrectly scaled, which produced distorted or near-zero warped outputs. This change applies the correct normalization so the TPS warp aligns with `grid_sample`’s expected coordinate system.

---

## 🛠️ Changes Made
- [x] convert TPS output from pixel coordinates → normalized coordinates for `grid_sample`.

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** The existing TPS warp tests pass without modification.
- [x] **Manual Verification:** 
  - I generated a simple synthetic image and applied a TPS warp.
  -  Before this change, the warped image output was nearly zero.
  - After normalization, the warp behaves as expected and preserves intensity.

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

### CODE REVIEW

**Comment by dheeraj25406** [comment]
Added, let me know if any other changes are required.
